From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 09:24:04 -0700
Subject: [PATCH] mm/damon/Kconfig: defaults VADDR, PADDR, and SYSFS to DAMON

It is suggested to enable DAMON_VADDR, DAMON_PADDR and DAMON_SYSFS
configurations together with DAMON, since those are essential parts of
DAMON for optimum and usual usages.  Because those need to be enabled
one by one, and there are other test-purpose or non-essential
configurations, it is easy to be confused and make mistakes at setup.
Make the essential configuration defaults to CONFIG_DAMON, so that those
can be enabled by default with single change.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index c213cf8b5638..c93d0c56b963 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -28,6 +28,7 @@ config DAMON_VADDR
 	bool "Data access monitoring operations for virtual address spaces"
 	depends on DAMON && MMU
 	select PAGE_IDLE_FLAG
+	default DAMON
 	help
 	  This builds the default data access monitoring operations for DAMON
 	  that work for virtual address spaces.
@@ -36,6 +37,7 @@ config DAMON_PADDR
 	bool "Data access monitoring operations for the physical address space"
 	depends on DAMON && MMU
 	select PAGE_IDLE_FLAG
+	default DAMON
 	help
 	  This builds the default data access monitoring operations for DAMON
 	  that works for the physical address space.
@@ -55,6 +57,7 @@ config DAMON_VADDR_KUNIT_TEST
 config DAMON_SYSFS
 	bool "DAMON sysfs interface"
 	depends on DAMON && SYSFS
+	default DAMON
 	help
 	  This builds the sysfs interface for DAMON.  The user space can use
 	  the interface for arbitrary data access monitoring.
-- 
2.39.5

